src/main/java/array/MainObject_Advance.java

```diff
@@ -173,35 +173,26 @@ public static void main(String[] args) {
         }
     }
 
-
+    //<editor-fold desc="Private functions">
     private static void showInfoBasic(Account account) {
         printSeparatedLine(50, "=");
-        System.out.println("Ho ten: " + account.getFirstName() + " " + account.getLastName());
-        System.out.println("Ngay tao: " + formatDate(account.getCreatedDate()));
+        System.out.println("Họ tên: " + account.getFirstName() + " " + account.getLastName());
+        System.out.println("Ngày tạo: " + formatDate(account.getCreatedDate()));
         System.out.println("Số tiền: " + formatCurrency(account.getCurrentBalance()));
         System.out.println("Loại kỳ hạn: " + account.getPeriod());
     }
 
-    private static void showInfo(Account account){
-     //   printSeparatedLine(50, "=");
-        System.out.println("Ho ten: " + account.getFirstName() + " " + account.getLastName());
-        System.out.println("Ngay tao: " + formatDate(account.getCreatedDate()));
-        System.out.println("Số tiền: " + formatCurrency(account.getCurrentBalance()));
-        System.out.println("Loại kỳ hạn: " + account.getPeriod());
-        System.out.println("Lịch sử giao dịch: ");
-        if (account.getHistories().size() == 0) {
-            System.out.println("Tài khoản chưa thực hiện bất kì giao dịch nào");
+    private static void showInfo(Account account, AccountHistory.Type type) {
+        List<AccountHistory> listAccountHistory = account.getHistories().stream().filter(transferIn ->
+                transferIn.getType() == type).collect(Collectors.toList());
+        showInfo(account, listAccountHistory);
```

Comment:
Do lúc này mình chỉ muốn in ra những lịch sử giao dịch với loại là **nhận tiền** thì khi gọi tới hàm showInfo(Account account, List<AccountHistory> listAccountHistory) mình nên **prepare** cái listAccountHistory trước bằng cách filter những accountHistory có type là **nhận tiền** (transferIn) trước.